### ISSUE TITLE
False positive: multiline math expressions detected as lists

### ISSUE THREAD

**Issue by vEnhance** [rvben/rumdl#556]
### rumdl version

0.1.64

### Installation method

Pre-built binary

### Bug description

In multiline math mode (`$$ ... $$`), it's common for lines to start with + or - for complicated multiline expressions. `rumdl` incorrectly parses these as list items, causing several different errors. An example of a file `math.md` that would trigger the errors:

```markdown
# Example math

## Plus signs

Using this and with some purely algebraic manipulations (namely, splitting the
integral into two parts, $0 \le x \le 1$ and $x \ge 1$, and then using the
property of the theta function), one can derive that

$$
\pi^{-s/2}\Gamma(s/2)\zeta(s) = \frac1{s-1} - \frac 1s
  + \int_1^\infty \frac{\theta(x)-1}{2} \cdot \left( x^{s/2-1} + x^{(1-s)/2-1} \right) dx.
$$

## Minus signs

To be explicit, we now have

$$
- \operatorname{Re} \frac{L'(s, \chi)}{L(s, \chi)}
  = \frac{1}{2} \log\frac{q}{\pi}
    + \frac{1}{2} \operatorname{Re} \frac{\Gamma'(\frac{1}{2} s + \frac{1}{2} a)}{\Gamma(\frac{1}{2} s + \frac{1}{2} a)}
    - \sum_{\rho} \operatorname{Re} \frac{1}{s-\rho} + \frac{\delta(\chi)}{s} + \frac{\delta(\chi)}{s-1}
$$
```

### Steps to reproduce

```
$ ls
math.md

$ rumdl check math.md --no-config
math.md:11:1: [MD005] Expected indentation of 0 spaces, found 2 [*]
math.md:11:1: [MD032] List should be preceded by blank line [*]
math.md:11:3: [MD004] List marker '+' does not match expected style '-' [*]
math.md:11:81: [MD013] Line length 90 exceeds 80 characters
math.md:19:1: [MD032] List should be preceded by blank line [*]
math.md:21:1: [MD007] Expected 2 spaces for indent depth 1, found 4 [*]
math.md:21:5: [MD004] List marker '+' does not match expected style '-' [*]
math.md:21:81: [MD013] Line length 120 exceeds 80 characters
math.md:22:1: [MD007] Expected 2 spaces for indent depth 1, found 4 [*]
math.md:22:81: [MD013] Line length 104 exceeds 80 characters

Issues: Found 7 issues in 1 file (24ms)
Run `rumdl fmt` to automatically fix 7 of the 7 issues
```

### Expected behavior

In principle, `rumdl` should detect that it's in a math environment and disable list-based rules.
(The MD013 errors are not false positives, though.)

### Actual behavior

False positives for rules like MD004, MD005, MD007, MD032, ...

### Configuration

```toml

```

### Operating system

Linux

**Comment by vEnhance** [comment]
#289 may be relevant, where link patterns were ignored inside math environments as well

**Comment by rvben** [comment]
Fixed in v0.1.65. Lines inside `$$ ... $$` display math blocks are now excluded from list detection, so operators like `-` and `+` no longer trigger MD004, MD005, MD007, or MD032 false positives. The fix also covers math blocks inside blockquotes (`> $$`).

Thanks for reporting, @vEnhance!
